lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -236,22 +467,48 @@ if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
 }
 [Net.ServicePointManager]::SecurityProtocol = $security_protocols
 
+# Check for case $checksum variable contain url. If yes, get file data from url and replace original value in $checksum
+if ($checksum_url) {
+
+    if ($checksum_url.startswith('http://', 1) -or $checksum_url.startswith('https://', 1) -or `
+        $checksum_url.startswith('ftp://', 1) -or [bool]([System.Uri]$checksum_url).isFile) {
+
+        $checksum = Get-ChecksumFromUrl -module $module -url $checksum_url -credentials $credentials `
+                                                -headers $headers -timeout $timeout -use_proxy $use_proxy `
+                                                -proxy $proxy -src_file_url $url
+    } else {
+        $module.FailJson("Unsupported `checksum_url` value for '$dest': '$checksum_url'")
+    }
+}
+
+
 if ($force -or -not (Test-Path -LiteralPath $dest)) {
 
-    Download-File -module $module -url $url -dest $dest -credentials $credentials `
+    Invoke-DownloadFile -module $module -url $url -dest $dest -credentials $credentials `
                   -headers $headers -timeout $timeout -use_proxy $use_proxy -proxy $proxy
 
 } else {
 
-    $is_modified = CheckModified-File -module $module -url $url -dest $dest -credentials $credentials `
-                                      -headers $headers -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+    $is_modified = Compare-ModifiedFile -module $module -url $url -dest $dest -credentials $credentials -headers $headers `
+                                      -timeout $timeout -use_proxy $use_proxy -proxy $proxy -check_length $false
 
     if ($is_modified) {
 
-        Download-File -module $module -url $url -dest $dest -credentials $credentials `
-                      -headers $headers -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+       Invoke-DownloadFile -module $module -url $url -dest $dest -credentials $credentials `
+                     -headers $headers -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+   } else {
+       $module.Result.msg = 'file already exists'
```

Comment:
I don't think this is necessary.

Reply:
It's needed for module Idempotence.
Could you suggest nay improvement?

Reply:
I don't understand why, `msg` is set in the documentation as the HTTP status message or an error message so this will be overwritten in a normal use case.

Reply:
Fixed